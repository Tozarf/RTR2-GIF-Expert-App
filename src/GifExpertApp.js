import React from 'react'
import { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'


export const GifExpertApp = () => {
    
    const [categories, setCategory] = useState(["Shingeki"])
   
    
    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategory={setCategory}/>
        <hr/>
        
        <ol>
        {
            categories.map((category)=>(
                <GifGrid
                key={category} 
                category={category}
                 />
        ))
        }
        </ol>
        </>
    )
}

