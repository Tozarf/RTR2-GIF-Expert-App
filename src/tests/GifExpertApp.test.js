import React from 'react'
import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'

describe('Testing <GifExpertApp/>', () => {
    test('should match the snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp/>)

        expect(wrapper).toMatchSnapshot()

    })

    test('should show a list of categories', () => {
        
        const categories = ["shingeki", "los simpsons"];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find("GifGrid").length).toBe(categories.length);


    })
    
    
})
