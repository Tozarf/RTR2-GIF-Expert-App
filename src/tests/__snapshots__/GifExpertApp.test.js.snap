// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp/> should match the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Testing <GifExpertApp/> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="shingeki"
      key="shingeki"
    />
    <GifGrid
      category="los simpsons"
      key="los simpsons"
    />
  </ol>
</Fragment>
`;
